### PR TITLE
Add process instance migration endpoint

### LINKED ISSUE #21208: Migrate the Migrate Process Instance gRPC API to REST
## Description

* The Camunda 8 REST API contains a resource deployment endpoint at `POST /v2/process-instances/{processInstanceKey}/migration` providing the same functionality as the [gRPC endpoint](https://docs.camunda.io/docs/next/apis-tools/zeebe-api/gateway-service/#migrateprocessinstance-rpc).
* The endpoint is featured in the OpenAPI description as (internally) [described](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.mysb5vgk5rx0).

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR add the process instance migration endpoint

## Related issues

closes #21208

### CODE REVIEW

**Review by tmetzke — APPROVED** [review]
👍 Fantastic job 🦕 
🔧 Just one minor naming suggestion

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -2987,6 +3036,41 @@ components:
...
+      type: object
+      properties:
+        targetProcessDefinitionKey:
+          description: The key of process definition to migrate the process instance to.
+          type: integer
+          format: int64
+        mappingInstructions:
+          type: array
+          items:
+            $ref: "#/components/schemas/MigrateProcessInstanceMappingInstructionRequest"
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3049]: Yes, sorry for the naming confusion

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -2987,6 +3036,41 @@ components:
...
+      required:
+        - targetProcessDefinitionKey
+        - mappingInstructions
+    MigrateProcessInstanceMappingInstruction:
+      type: object
+      description: |
+        The mapping instructions describe how to map elements from the source process definition to the target process definition.
+      properties:
+        sourceElementId:
+          description: The element id to migrate from.
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3066]:
```suggestion
          description: The element ID to migrate from.
```

🔧  I defer to a technical writer's review on this, but I think this is the way we capitalize ID. 

💭 It was also mentioned in [another PR](https://github.com/camunda/camunda/pull/22137#discussion_r1752288711) that [the guidance I wrote up](https://github.com/camunda/camunda/blob/main/docs/rest-controller.md#openapi-extension) suggested that id should not be capitalized. If ID should be capitalized, I will update that guidance to hopefully remove that confusion.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -2987,6 +3036,41 @@ components:
...
+          type: string
+        targetElementId:
+          description: The element id to migrate into.
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3069]:
```suggestion
          description: The element ID to migrate into.
```

🔧 Same as previous comment.

**Review by pepopowitz — APPROVED** [review]
🔧 A couple very small suggestions on capitalization, which I defer to a technical writer on.

👍 Otherwise it looks great!

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 2 above)
Comment (tmetzke) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3066]: The [naming conventions of the C8 API guidelines](https://docs.google.com/document/d/1G9AmmNac-4QLGZ0LXQXa3FyeCrSJdIboaPt3-R6dWNw/edit#heading=h.8kew5enz4jtm) (based on guidelines created earlier by DevEx) suggest to use lowercase.

Comment (conceptualshark) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3066]: Naming conventions or code references aren't quite the same as uses in text; our fallback is the Google style guide, which also suggests capitalization. It's something currently under discussion, though, so there's no set internal standard either way.

Comment (nicpuppa) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3066]: To be consistent within the specification I will use uppercase for ID

**Pull request merged**